### PR TITLE
Have Docs/Reference sidebars reference each other

### LINKED ISSUE #8: Consider showing both text docs "Docs" and API docs "Reference" in single sidebar, same as old site
From #4:

It seems a bit confusing for users to split the long-form text ("Docs") and detailed API descriptions ("Reference") into separate categories as they are: if I'm looking for how to learn something, I'd be hoping that everything I need is easy to find. Alternatives:
   - Include the reference sections in the sidebar for the docs view (maybe with a divider, if that's possible), but still have a direct link to it in the headings. That is, these things could be after "Tutorials - Coming soon"
      image
   - If that doesn't work great, just have a link to "Reference" at the end of the Docs sidebar too, and a link to "Docs" in the "Reference" sidebar. (I.e. repeating links from the header, but in the place users might be looking)

### PR DESCRIPTION
**Pull request by thejcannon**
Fixes #8 by adding "cross-links" from/to Docs/Reference at the bottom of their respective sidebars.

### CODE REVIEW

File: sidebars.js [hunk 1]
```diff
@@ -1,6 +1,30 @@
...
+  referenceSidebar: [
+    { type: "autogenerated", dirName: "reference" },
+    {
+      type: "html",
+      value: "<hr/>",
+      className: "margin-vert--none",
+    },
+    {
+      type: "doc",
+      id: "docs/introduction/how-does-pants-work",
```
Comment (huonw) [inline sidebars.js:24]: This doesn't need to be solved now, but just exploring the space: what happens if we want to change this entry page?

For instance, we add some awesome new landing page for Pants 2.34, that doesn't exist in 2.33 or earlier.

Is it possible to customise per version? Or use some generic entry point like `docs` that each version redirects itself?

**Review by huonw — APPROVED** [review]

File: sidebars.js (hunk 1 above)
Comment (thejcannon) [reply sidebars.js:24]: Yes, there is per-version sidebars (this particular file is for `docs/`, note all the other files in this PR (one per version)

Comment (huonw) [reply sidebars.js:24]: Oh, From the file names, I assumed they were auto-generated from this one... are they all manually maintained? When we add a new version do we have to manually copy it across?

Comment (thejcannon) [reply sidebars.js:24]: Nope, it's handled automatically by the versioning scripts. Check out the Docusaurus versioning docs

Comment (huonw) [reply sidebars.js:24]: Ah, I see https://docusaurus.io/docs/versioning#tagging-a-new-version

Once it's tagged, I assume they're then manually maintained? As in, you made these edits "manually" (by copying the file around, I imagine)?

Comment (thejcannon) [reply sidebars.js:24]: Yup! That's right

**Pull request merged**